File: agents/ocean/uncertainty.py
def assess_uncertainty(confidence):
    """
    Convert confidence into an uncertainty level.
    Higher confidence means lower uncertainty.
    """

    if confidence is None:
        return {
            "level": "HIGH",
            "reason": "Confidence data is unavailable"
        }

    if confidence >= 0.85:
        return {
            "level": "LOW",
            "reason": "Marine data confidence is high"
        }

    if confidence >= 0.70:
        return {
            "level": "MEDIUM",
            "reason": "Marine data confidence is moderate"
        }

    return {
        "level": "HIGH",
        "reason": "Marine data confidence is low"
    }

File: src/utils/uncertainty.js
export function getUncertaintyLevel(score) {
  if (score === null || score === undefined) {
    return "INSUFFICIENT";
  }

  if (score <= 20) {
    return "LOW";
  }

  if (score <= 40) {
    return "MEDIUM";
  }

  return "HIGH";
}